### DISCUSSION TITLE
A decent guide to using terracognita

### DISCUSSION THREAD

**Discussion by dannous** [cycloidio/terracognita#324] [General]
Hi, is there a decent guide for using terracognita? the readme provides bare minimum information about it and id love to know more about how to use it and how seperating into modules and state files is supposed to work.
I use AWS and am looking for the equivalent of a novice step by step guide to export a large amount of existing infrastructure as terraform, and separate it into modules & variables (which it looks like terracognita can do)

**Comment by vcolombo** [comment]
I would love better examples as well, specifically for modularising existing environments. I'm able to output everything as a single, monolithic module using Terracognita, but that defeats the purpose of using modules. Is there a way to run Terracognita that creates per-service modules and a correctly matching state file? I've tried running Terracognita on each service in our infrastructure, outputting it as individual modules, but that doesn't match the state file and causes errors when running a plan.
